### ISSUE TITLE
Feature Requset: Callback for external trigger, before sensor read-off

### ISSUE THREAD

**Issue by kmcgrathgenerate** [TheImagingSource/tiscamera#539]
**Is your feature request related to a problem? Please describe.**
I am using the TIS camera to act as a driver for a two-camera system. Currently, I trigger the TIS camera, get the image in the framereadycallback, and then get the second cameras image. They are both triggered but the function to get the second cameras image takes a long time and is unavoidable. 

**Describe the solution you'd like**
Rather than have them in serial, I was hoping I could start the second cameras getter function faster if I knew the TIS camera was triggered before the camera handles its sensor read-off / sending to the frameready callback

In short, is there a way I can see immediately that the TIS camera was triggered with some callback?

**Comment by kmcgrathgenerate** [comment]
I am using the DMK 37BUX287 specifically with the python wrapper

**Comment by TIS-Stefan** [comment]
Hello

Well, the best solution for synchronizing he cameras is connected both to the same trigger source. In case you use hardware trigger.
In case you use software trigger, the best method I know. is creating a thread for each camera. In the thread the software trigger functions is called, nothing more, the thread function can return. This avoids a delay between cameras, because the software trigger send function waits for an answer of the camera. If both software triggers run in "parallel", this waiting time can be avoided.

Stefan

**Comment by kmcgrathgenerate** [comment]
Yes I agree, and that is what we are currently doing. The issue is that the SDK for the other camera we are using is lacking and has no way to let us know when a new image is taken. There is no callback or any indication that an image we grab with their 'getter' is new or not (without doing some sort of image subtraction with the previous image). 

That's why I was hoping for a way to know when the trigger was in earlier so I could do parallel processing instead of serial like we are forced to do for now.

**Comment by TIS-Stefan** [comment]
GStreamer uses a callback from the appsink module for incoming images. So you can get the images from there and start your processing. That is, how the trigger samples in this repository work.

**Comment by TIS-Edgar** [comment]
Closed due to inactivity.
